modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleArtifactContext.java

```diff
@@ -285,39 +284,81 @@ protected void loadBeanDefinitions(DefaultListableBeanFactory beanFactory) throw
       }
     } finally {
       currentMuleContext.remove();
     }
   }
 
+  private void createComponents(DefaultListableBeanFactory beanFactory, ApplicationModel applicationModel, boolean mustBeRoot) {
+    List<String> createdComponentModels = new ArrayList<>();
+    applicationModel.executeOnEveryMuleComponentTree(componentModel -> {
+      if (!mustBeRoot || componentModel.isRoot()) {
+        if (componentModel.getIdentifier().equals(MULE_IDENTIFIER)) {
+          return;
+        }
+        if (componentModel.getNameAttribute() != null) {
+          createdComponentModels.add(componentModel.getNameAttribute());
+        }
+        beanDefinitionFactory.resolveComponentRecursively(applicationModel.getRootComponentModel(), componentModel,
+                                                          beanFactory,
+                                                          (resolvedComponentModel, registry) -> {
+                                                            if (resolvedComponentModel.isRoot()) {
+                                                              String nameAttribute =
+                                                                  resolvedComponentModel.getNameAttribute();
+                                                              if (resolvedComponentModel.getIdentifier()
+                                                                  .equals(CONFIGURATION_IDENTIFIER)) {
+                                                                nameAttribute = OBJECT_MULE_CONFIGURATION;
+                                                              } else if (nameAttribute == null) {
+                                                                // This may be a configuration that does not requires a name.
+                                                                nameAttribute = uniqueValue(resolvedComponentModel
+                                                                    .getBeanDefinition().getBeanClassName());
+                                                              }
+                                                              registry.registerBeanDefinition(nameAttribute,
+                                                                                              resolvedComponentModel
+                                                                                                  .getBeanDefinition());
+                                                              postProcessBeanDefinition(componentModel, registry,
+                                                                                        nameAttribute);
+                                                            }
+                                                          }, null);
+      }
+    });
+    applyLifecycle(createdComponentModels);
+  }
+
+  private void applyLifecycle(List<String> createdComponentModels) {
+    if (muleContext.isInitialised()) {
```

Comment:
I find a bit weird that the context is initialized but some components are not. And then you need to manually initiailze a component. I understand that this magic is required to avoid having to initialized the full context, but having this feature on the MuleArtifactContext makes some noise to me. Is there any chance to have a LazyMuleArtifactContext subclass?